e2e/pages/login.po.ts

```diff
@@ -0,0 +1,15 @@
+import { element, by } from 'protractor';
+
+export class LoginPage {
+    username = element(by.id('username'))
```

Comment:
Is it safe to initialize the element in this context? Should we find an element just before accessing it? (Namely: inside the 'login' method?) From the docs I can see:
'An ElementFinder does not actually attempt to find the element until an action is called'. But then will it try to find an element before EVERY action or only after the first one?

In this case it's all not a problem, I'm thinking if at some point some more dynamic elements will cause us trouble because we will reference an element that no longer exists on the page.

Reply:
>>> 'An ElementFinder does not actually attempt to find the element until an action is called'. But then will it try to find an element before EVERY action or only after the first one?


I'm not 100% sure if protractor search for an element before every action but as you wrote `An ElementFinder does not actually attempt to find the element until an action is called` so it's safe to declare it in this way.
There is also one more thing. We are using implicit wait and what is it: 
`This is special hidden automatic wait, on each driver.findElement(...). Original webdriver (js, python, java) throws NoSuchElementException if element cannot be found in page DOM structure. This kind of wait will be done before EVERY driver.findElement, no matter what kind of locator do you use. When implicit wait timed out, NoSuchElementException will be thrown outside findElement function.`

So taking these two into consideration, we should be safe even when dynamically we change DOM